src/v/kafka/server/partition.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cluster/partition.h"
+#include "model/fundamental.h"
+#include "storage/types.h"
+
+namespace kafka {
+
+/**
+ * Kafka layer wrapper for partition, this wrapper performs offsets translation
+ * to eliminate offset skew caused by existantece of Raft configuration batches.
+ */
+// TODO: virtualize to
+class partition {
```

Comment:
we already have cluster::partition and it's a bit confusing to reuse the same term for another entity; can you rename it please to anything unique e.g. partition_wrapper, proxy_partition, translating_partition etc.

Reply:
it is in different namespace `kafka::partition`, we already have few places like this. f.e. there is `kafka::fetch_session` and `kafka::client::fetch_session`, I prefer the namespace to serve as a part of name

Reply:
there is a distinct meaning between cluster::partition (raw offset) and kafka::partition (transformed offset) and since we don't have type distinctions between the offsets it's better to highlight this difference in the name of the types

also in general reusing a name leads to inconveniences. for example I tend to use `git grep` to explore code base and since we don’t specify a namespace on type declaration and tend to omit namespaces so when we reuse the type the search results become too noisy